src/common/config.h

```diff
@@ -108,24 +51,21 @@ class Config {
       return dflt;
   }
 
+  const YAML::Node& get() const;
+  YAML::Node& get();
+
   template <typename T>
   void set(const std::string& key, const T& value) {
     config_[key] = value;
   }
 
-  const YAML::Node& get() const;
-  YAML::Node& get();
-
   YAML::Node getModelParameters();
   void loadModelParameters(const std::string& name);
   void loadModelParameters(const void* ptr);
 
-  const std::vector<DeviceId>& getDevices() { return devices_; }
+  const std::vector<DeviceId>& getDevices();
```

Comment:
Why replace this function by a non-inlined one that does exactly the same thing? This makes the code harder to read, since the reader now must jump between two files. I suggest to revert this back.

Reply:
The class has been blindly separated into .h and .cpp, as it will be majorly refactorized later. Moreover, this method might be removed as suggested in another comment.